tests/operators/test_sftp_to_gcs.py

```diff
@@ -0,0 +1,213 @@
+            move_object=True,
+            gcp_conn_id=GCP_CONN_ID,
+            sftp_conn_id=SFTP_CONN_ID,
+            delegate_to=DELEGATE_TO,
+        )
+        task.execute(None)
+
+        sftp_hook.return_value.get_tree_map.assert_called_with(
+            "main_dir", prefix="main_dir/test_object", delimiter=".json"
+        )
+
+        sftp_hook.return_value.retrieve_file.assert_has_calls(
+            [
+                mock.call("main_dir/test_object3.json", mock.ANY),
+                mock.call("main_dir/sub_dir/test_object3.json", mock.ANY),
+            ]
+        )
+
+        gcs_hook.return_value.upload.assert_has_calls(
+            [
+                mock.call(
+                    bucket_name=TEST_BUCKET,
+                    object_name="destination_dir/test_object3.json",
+                    mime_type=DEFAULT_MIME_TYPE,
+                    filename=mock.ANY,
+                ),
+                mock.call(
+                    bucket_name=TEST_BUCKET,
+                    object_name="destination_dir/sub_dir/test_object3.json",
+                    mime_type=DEFAULT_MIME_TYPE,
+                    filename=mock.ANY,
+                ),
+            ]
+        )
+
+    @mock.patch("airflow.operators.sftp_to_gcs.GoogleCloudStorageHook")
+    @mock.patch("airflow.operators.sftp_to_gcs.SFTPHook")
+    def test_execute_move_with_wildcard(self, sftp_hook, gcs_hook):
+        sftp_hook.return_value.get_tree_map.return_value = [
+            ["main_dir/test_object3.json", "main_dir/sub_dir/test_object3.json"],
+            [],
+            [],
+        ]
+
+        gcs_hook.return_value.list.return_value = SOURCE_FILES_LIST[:2]
+        task = SFTPToGoogleCloudStorageOperator(
+            task_id=TASK_ID,
+            source_path=SOURCE_OBJECT_WILDCARD_FILENAME,
+            destination_bucket=TEST_BUCKET,
+            destination_path=DESTINATION_PATH_DIR,
+            move_object=True,
+            gcp_conn_id=GCP_CONN_ID,
+            sftp_conn_id=SFTP_CONN_ID,
+            delegate_to=DELEGATE_TO,
+        )
+        task.execute(None)
+
+        sftp_hook.return_value.delete_file.assert_has_calls(
+            [
+                mock.call("main_dir/test_object3.json"),
+                mock.call("main_dir/sub_dir/test_object3.json"),
+            ]
+        )
+
+    @mock.patch("airflow.operators.sftp_to_gcs.GoogleCloudStorageHook")
+    @mock.patch("airflow.operators.sftp_to_gcs.SFTPHook")
+    def test_execute_more_than_one_wildcard_exception(self, sftp_hook, gcs_hook):
+        task = SFTPToGoogleCloudStorageOperator(
+            task_id=TASK_ID,
+            source_path=SOURCE_OBJECT_MULTIPLE_WILDCARDS,
+            destination_bucket=TEST_BUCKET,
+            destination_path=DESTINATION_PATH_FILE,
+            move_object=False,
+            gcp_conn_id=GCP_CONN_ID,
+            sftp_conn_id=SFTP_CONN_ID,
+            delegate_to=DELEGATE_TO,
+        )
+        with self.assertRaises(AirflowException):
```

Comment:
Would be great to check the exception message here. We had in the past problems that different exception was raised and we did not realise it.

Reply:
Good point. I will add check here.